### PR TITLE
feat: FSA

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/collectors/rest/plugins/volumeanalytics/volumeanalytics.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+		}
+	}
+	return nil
+}
+
+func (v *VolumeAnalytics) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
+
+	cluster, _ := data.GetGlobalLabels().GetHas("cluster")
+	ver := v.client.Cluster().Version
+	clusterVersion := fmt.Sprintf("%d.%d.%d", ver[0], ver[1], ver[2])
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/volumeanalytics/volumeanalytics.go:94]: minor but we do this in two places - probably worth adding a method to `Cluster` that does this and refactor both to us

File: cmd/collectors/rest/plugins/volumeanalytics/volumeanalytics.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+	if err != nil {
+		v.Logger.Warn().Err(err).Msg("error while init matrix")
+		return nil, err
+	}
+	for k := range v.data {
+		// Set all global labels if already not exist
+		v.data[k].SetGlobalLabels(data.GetGlobalLabels())
+	}
+
+	if ontapVersion.GreaterThanOrEqual(version98After) {
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/volumeanalytics/volumeanalytics.go:121]: is this check needed since we checked < at line 105?

**Review by cgrinds — COMMENTED** [review]
nice job!

File: pkg/util/util.go [hunk 3]
```diff
@@ -350,3 +352,42 @@ func ParseZAPIDisplay(obj string, path []string) string {
...
+	}
+	v := u.Query()
+	mr := v.Get(query)
+	return mr, nil
+}
+
+func EncodeURL(href string) string {
+	u, err := url.Parse(href)
+	if err != nil {
+		fmt.Println(err)
```
Comment (cgrinds) [inline pkg/util/util.go:389]: debug code left?

**Review by cgrinds — APPROVED** [review]

**Review by Hardikl — APPROVED** [review]

**Pull request merged**